<a href="https://colab.research.google.com/github/edergomezhe/Fudnamentos_DS/blob/main/Challenge_Depresion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import scipy.stats.distributions as dist
import seaborn as sns
import scipy.stats as stats
from collections import Counter
import matplotlib.pyplot as plt

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


# Challenge

El [Instituto Nacional de Salud Mental](https://www.nimh.nih.gov/) publicó un [artículo](https://www.nimh.nih.gov/health/statistics/major-depression) que indica que en cualquier período de un año, 
aproximadamente el **9.5 por ciento de los adultos estadounidenses** sufren depresión o una enfermedad depresiva.
Supongamos que en una **encuesta de 100 personas en una determinada ciudad**, **siete de ellas** sufrieron depresión 
o una enfermedad depresiva.

Realice una prueba de hipótesis para determinar si la verdadera **proporción de personas en esa ciudad** 
que padecen depresión o una enfermedad depresiva **es menor** que el porcentaje de la **población estadounidense** 
adulta en general.

# Resuelve siguiendo los pasos a continuación

## Paso 0:

- ¿Es esta una prueba de una media o proporción?



**Es una prueba de proporción**

## Paso 1: Determinar la hipótesis nula y las hipótesis alternativas

- Enuncie las hipótesis nula y alternativa. Hágalo como un enunciado y como una expresión matemática.

H0: La proporcion de personas con depresion en la ciudad es mayor a la poblacion estadounidense p_pob > p0

H1: La proporcion de personas con depresion en la ciudad es menor a la poblacion estadounidense p_pob < p0

## Paso 2: Calcular la proporción de la población

In [2]:
#calculo de proporcion de la poblacion que sufrio de depresion o de alguna enfermedad depresiva segun la encuesta
p_pob = 7/100

#imprimir la proporcion
p_pob

0.07

## Paso 3: Calculando el test estadístico

In [3]:
#porcentaje aproximado de los adultos estadounidenses sufren depresión o una enfermedad depresiva.
p0 = 0.095

In [4]:
#calcular el error estándar (SE).
n = 100

se = np.sqrt(p0 * (1 - p0) / n)

#Imprimir el valor
se

0.0293214938227915

In [5]:
#Calculamos la mejor estimación (best estimate - be)
be = p_pob

# hypothesized estimate
he = p0

# Calculamos la estadística de prueba (Test Stat)
test_stat = (be - he)/se

#Imprimir el valor
test_stat

-0.8526168602149314

## Paso 4: Calcule el valor p

In [6]:
#Calcular el valor p.
pvalue = 2*dist.norm.cdf(-np.abs(test_stat))

#imprimir el valor
pvalue

0.39387180961069246

## Paso 5: inferir la conclusión del valor p

Si el nivel de significancia es α = 0.05, cuál es tu:

- Decisión: La hipotesis nula no se puede rechazar con suficiente nivel de confiabilidad

- Razón de la decisión: El nivel de significancia es de un 39.4% por lo tanto hay una probabilidad del 60.6% de que los resultados no sean aleatorios

- Conclusión (escribir en una oración completa): A pesar de que aparentemente existe un menor porcentaje de personas que sufren enfermedades mentales en esa ciudad, no es posible aseverar contundentemente que la muestra tomada de dicha ciudad tenga la proporcion real de enfermedades mentales en esa ciudad. Por lo tanto no podemos comprobar que la proporcion de enfermedades mentales en esa ciudad sea menor a la proporcion de toda la poblacion estadounidense.

- PREGUNTA BONUS: ¿Es esta una prueba de cola derecha, izquierda o de dos colas?

Es una prueba de cola izquierda ya que la hipotesis alternativa tiene el signo < en su enunciado.